Sources/FaunaDB/Expr.swift

```diff
@@ -1,5 +1,47 @@
 import Foundation
 
+/**
+    `Expr` is the top level of the query language expression tree.
+
+    For convenience, some native types are considered valid expressions:
+    `String`, `Int`, `Double`, `Bool`, `Optional`, and `Date`.
+
+    ## Motivation
+
+    The FaunaDB query language is a non-typed expression based language.
+    You can think about it as an actual programming language.
+
+    In order to express the whole power and flexibility of the query language,
+    we need to express our DSL in a way that allows the same combinations as a
```

Comment:
I'm not sure what this is trying to say; can you restate it a bit to help clarify?